### DISCUSSION TITLE
ezdxf Bin-Packing Add-on remove the item that do not stack on top of each other

### DISCUSSION THREAD

**Discussion by mjsandoval04** [mozman/ezdxf#802] [Q&A]
Hello folks,

I am new to ezdxf, so go easy on me.

I've been trying to use ezdxf Bin-Packing Add-on to create a simulation in Anylogic in which operators use a pallet-trolley to pick boxes/items in a specific order from a warehouse.  The pick order is based on the results of the Bin-Packing algorithm. 

the container/bin is based on the standard dimension of a EuroPallet and the boxes are standard carton boxes with different sizes. 

My issue is that some of the boxes do not overlap 100%.  This is ok if the pallet is actually enclosed but in a real-world scenario like using a pallet trolley to pick boxes if the boxes are not put one on top of the other they will probably fall down

my question is there a way to remove the items/boxes with surfaces that does not overlap with another box?

Current results of running the Bin-Packing Add-on

![boxesissue2 DXF_001](https://user-images.githubusercontent.com/92613179/208008082-dd630fcc-11ed-4c54-b842-27dbca181d1f.png)

Desire results after removing items/boxes with the surface not overlapping 100% another item/box (the remaining boxes should be packed in another container/bin)

![boxes issue 2 DXF](https://user-images.githubusercontent.com/92613179/208008302-1c9b9afb-006d-4924-abec-1009fd94e0ec.png)

**Comment by mozman** [comment]
1. No, there exist no such specific function for this special case.

2. You shouldn't use an add-on from a package that has a completely different main purpose for real world problems - it might not be well tested for such use cases. The result clearly shows that this add-on is not suitable for your problem!

3. The check is a bit more complex: for each two boxes check if they overlap in plan (top) view: yes -> if their top (bottom) and bottom (top) plan do not touch remove the top box and also remove all boxes above the removed box.

EDIT: putting the result in a physic engine may improve the result,  but using an appropriate algorithm that takes physics into account is certainly the better way.
